libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,443 @@
+  filter,
+  map,
+  observeOn,
+  share,
+  switchMap,
+  takeUntil,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import {
+  getKeyCodeValue,
+  getSliderPositionBasedOnValue,
+  getSliderValueForCoordinate,
+  roundToSnap,
+} from './slider-utils';
+
+/** We need unique ids in order to have correct labeling. */
+let uniqueId = 0;
+
+const KEY_CODES_ARRAY: Array<number> = [
+  LEFT_ARROW,
+  DOWN_ARROW,
+  RIGHT_ARROW,
+  UP_ARROW,
+  HOME,
+  END,
+  PAGE_UP,
+  PAGE_DOWN,
+];
+
+declare const window: any;
+
+@Component({
+  selector: 'dt-slider',
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.scss'],
+  host: {
+    class: 'dt-slider',
+  },
+  encapsulation: ViewEncapsulation.Emulated,
+  exportAs: 'dtSlider',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtSlider implements AfterViewInit, OnDestroy, OnInit {
+  /**
+   * Holds the value, the rounding is shifted with.
+   * Calculated based in the step, to avoid JS rounding problems.
+   */
+  private _roundShift: number = 0;
+
+  /** @internal Unique id for this input. */
+  _labelUid = `dt-slider-label-${uniqueId++}`;
+
+  /** Holds the value of the slider. */
+  private _value$ = new BehaviorSubject<number>(0);
+  /** Holds the description of the size of the slider. */
+  private _clientRect$: Observable<ClientRect>;
+  /** Observer that gets triggered if the slider is resized on the screen. */
+  private _resizeObserver$ = new Subject();
+  /** Variable to hold the ResizeObserver */
+  private _observer: any;
+  /** Observer that gets triggered if the input field value is changed. */
+  private inputFieldValue$ = new Subject<number>();
+
+  /**
+   * Binding for the minimum value of the slider.
+   * TODO: getter and setter
+   */
+  @Input()
+  get min(): number {
+    return this._min;
+  }
+  set min(min: number) {
+    this._min = min;
+    this._updateSliderPosition(this._value, this._min, this._max);
+  }
+  private _min: number;
+
+  /**
+   * Binding for the maximum value of the slider.
+   * TODO: getter and setter
+   */
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+  set max(max: number) {
+    this._max = max;
+    this._updateSliderPosition(this._value, this._min, this._max);
+  }
+  private _max: number;
+
+  /**
+   * Bindings for the step, if changed, roundShift needs to be recalculated.
+   */
+  @Input()
+  get step(): number {
+    return this._step;
+  }
+  set step(step: number) {
+    if (isDefined(step)) {
+      this._step = step;
+    }
+
+    this._roundShift = 0;
+    // deal with rounding problems.
+    let countingStep = this._step;
+
+    while (countingStep % 1 !== 0) {
+      countingStep *= 10;
+      this._roundShift++;
+    }
+  }
+  /** Holds the value of step internally */
+  private _step: number = 5;
+
+  /** Binding for the disabled state. */
+  @Input()
+  get disabled(): boolean {
+    return this._isDisabled;
+  }
+  set disabled(disabled: boolean) {
+    this._isDisabled = disabled;
```

Comment:
```suggestion
    this._isDisabled = coerceBooleanProperty(disabled);
```

If we use `coerceBooleanProperty` from the cdk here, we will enable consumers to not have to assign the disabled in the template, because this also checks if it is at all defined.

`<dt-slider disabled></dt-slider>` will then work instead of requiring `<dt-slider disabled=true></dt-slider>`

Reply:
Awesome, I will update it.